src/ivc/step_circuit.rs

```diff
@@ -123,20 +123,21 @@ pub(crate) struct StepInputs<'link, const ARITY: usize, C: CurveAffine, RO: ROCi
 where
     C::Base: ff::PrimeFieldBits + ff::FromUniformBytes<64>,
 {
-    public_params: &'link SynthesizeStepParams<C, RO>,
+    step_public_params: &'link SynthesizeStepParams<C, RO>,
```

Comment:
Here use  "link" instead of "a", what does link stand for

Reply:
A one-letter annotation is bad practice for lifetime, it's just one of the generic parameters and it may well be telling.

Reply:
Why it is bad practice? I have seen it been used in many libraries.  And I still don't get the meaning of "link" here yet